src/bp/ui-admin/src/Pages/Bot/Details.jsx

```diff
@@ -124,8 +137,27 @@ class Bots extends Component {
     )
   }
 
+  cancel = () => {
+    if (this.state.formHasChanged) {
+      // [TODO] max.cloutier 2020.01.17 Implement and use a custom confirm popup and replace the window.confirm in this file/app-wide
+      const conf = window.confirm(`There are unsaved changes in this form. Are you sure you want to cancel?`)
+
+      if (conf) {
+        this.backToList()
+      }
+    } else {
+      this.backToList()
+    }
+  }
+
+  backToList = () => {
+    const workspaceId = getActiveWorkspace()
+
+    this.props.history.push(generatePath('/workspace/:workspaceId?/bots', { workspaceId: workspaceId || undefined }))
```

Comment:
By default, when you visit /workspace, it will automatically update the url with the correct workspace ID. Simply calling `this.props.history.push('/workspace')` would do the trick